projects/plugins/jetpack/modules/videopress/class-videopress-attachment-metadata.php

```diff
@@ -0,0 +1,159 @@
+<?php
+/**
+ * Handle the VideoPress metadata properties.
+ *
+ * @package Jetpack
+ */
+
+use Automattic\Jetpack\Connection\Client;
+
+/**
+ * Class Videopress_Attachment_Metadata
+ */
+class Videopress_Attachment_Metadata {
+
+	/**
+	 * Persist the VideoPress metadata information, including rating and display_embed.
+	 *
+	 * @param string|int $post_id       The post id.
+	 * @param string     $guid          VideoPress Guid.
+	 * @param string     $post_title    The post title.
+	 * @param string     $caption       Video caption.
+	 * @param string     $post_excerpt  The post excerpt.
+	 * @param string     $rating        The rating.
+	 * @param int        $display_embed The display_embed.
+	 *
+	 * @return bool|\WP_Error
+	 */
+	public static function persist_metadata( $post_id, $guid, $post_title, $caption, $post_excerpt, $rating, $display_embed ) {
+		$post_id = absint( $post_id );
+
+		$args = array(
+			'method'  => 'POST',
+			'headers' => array( 'content-type' => 'application/json' ),
+		);
+
+		$values         = self::build_wpcom_api_request_values( $post_title, $caption, $post_excerpt, $rating, $display_embed );
+		$endpoint       = 'videos';
+		$values['guid'] = $guid;
+
+		$result = Client::wpcom_json_api_request_as_blog( $endpoint, '2', $args, wp_json_encode( $values ), 'wpcom' );
```

Comment:
In the old implementation we used the v1.1 rest API endpoint (/videos/{guid}) for updating the VideoPress metadata stored in WPCOM. While this worked fine for Jetpack and Atomic, when executed in WPCOM context the route returned a 404 error message, saying that the route doesn't exist.

I've made some debugging in WPCOM with xDebug and noticed that the routes for v1.1 weren't registered, for some reason. To by-pass this issue in WPCOM and to have the same logic for both environments (WPCOM and Atomic), I've changed the endpoint to the new V2 endpoint (POST /videos/)